plugin-build/src/main/kotlin/io/sentry/android/gradle/telemetry/SentryTelemetryService.kt

```diff
@@ -0,0 +1,474 @@
+                            "in the sentry config block."
+                    )
+                    Sentry.init { options ->
+                        options.dsn = startParameters.dsn
+                        options.isDebug = true
+                        options.isEnablePrettySerializationOutput = false
+                        options.tracesSampleRate = 1.0
+                        options.release = BuildConfig.Version
+                        options.isSendModules = false
+                        options.environment = startParameters.buildType
+                        options.setTag("SDK_VERSION", BuildConfig.SdkVersion)
+                        options.setTag("AGP_VERSION", AgpVersions.CURRENT.toString())
+                        options.setTag("BUILD_SYSTEM", "gradle")
+                        options.setTag("BUILD_TYPE", startParameters.buildType)
+                        options.setTag("GRADLE_VERSION", GradleVersion.current().version)
+                        startParameters.cliVersion?.let { options.setTag("SENTRY_CLI_VERSION", it) }
+
+                        startParameters.extraTags.forEach { (key, value) ->
+                            options.setTag(
+                                key,
+                                value
+                            )
+                        }
+                    }
+                }
+                hub = Sentry.getCurrentHub()
+                startRun("gradle build ${startParameters.buildType}")
+
+                hub.configureScope { scope ->
+                    scope.user = User().also { user ->
+                        startParameters.defaultSentryOrganization.let { user.id = it }
+                        startParameters.sentryOrganization.let { user.id = it }
+                    }
+                }
+            }
+        } catch (t: Throwable) {
+            SentryPlugin.logger.error("Sentry failed to initialize.", t)
+        }
+    }
+
+    override fun started(descriptor: BuildOperationDescriptor, event: OperationStartEvent) {}
+
+    override fun progress(identifier: OperationIdentifier, event: OperationProgressEvent) {}
+
+    override fun finished(
+        buildOperationDescriptor: BuildOperationDescriptor,
+        operationFinishEvent: OperationFinishEvent
+    ) {
+        val details = buildOperationDescriptor.details
+
+        operationFinishEvent.failure?.let { error ->
+            // TODO check if Sentry task, otherwise ignore
+            if (isSentryError(error, details)) {
+                captureError(error, "build")
+                transaction?.status = SpanStatus.UNKNOWN_ERROR
+            }
+        }
+
+        if (details is RunRootBuildWorkBuildOperationType.Details) {
+            endRun()
+        }
+    }
+
+    private fun isSentryError(throwable: Throwable, details: Any?): Boolean {
+        val isSentryTaskName = (details as? ExecuteTaskBuildOperationDetails)
+            ?.let {
+                it.task.name.substringAfterLast(":")
```

Comment:
nit: we could also check for the task type here, not sure if it's more reliable, but this is also fine 👍

Reply:
Hmm if there's no need to check I'd just keep it as is for now.

Reply:
sure, feel free to ignore it